src/gmpints.c

```diff
@@ -375,6 +375,31 @@ Obj ObjInt_UInt( UInt i )
   }
 }
 
+Obj ObjInt_Int8( Int8 i )
+{
+  int neg = 0;
+  if(i < 0) {
+    neg = 1;
+    i = -i;
+  }
```

Comment:
Drat, I thought such a simple method would be "obviously correct". Obviously not!